wherehows-web/app/constants/dataset-compliance.ts

```diff
@@ -250,6 +252,31 @@ const mergeMappedColumnFieldsWithSuggestions = (
     return field;
   });
 
+/**
+ * Creates a map of compliance changeSet identifier field to compliance change sets
+ * @param {IIdentifierFieldWithFieldChangeSetObject} identifierFieldMap
+ * @param {IComplianceChangeSet} changeSet
+ * @returns {IIdentifierFieldWithFieldChangeSetObject}
+ */
+const foldComplianceChangeSetToField = (
+  identifierFieldMap: IIdentifierFieldWithFieldChangeSetObject,
+  changeSet: IComplianceChangeSet
+): IIdentifierFieldWithFieldChangeSetObject => ({
+  ...identifierFieldMap,
+  [changeSet.identifierField]: [...identifierFieldMap[changeSet.identifierField], changeSet]
+});
+
+/**
+ * Reduces a list of IComplianceChangeSet to a list of tuples with a complianceChangeSet identifierField
+ * and a changeSet list
+ * @param {Array<IComplianceChangeSet>} changeSet
+ * @returns {Array<IdentifierFieldWithFieldChangeSetTuple>}
+ */
+const foldComplianceChangeSets = (
+  changeSet: Array<IComplianceChangeSet>
+): Array<IdentifierFieldWithFieldChangeSetTuple> =>
+  Object.entries<Array<IComplianceChangeSet>>(arrayReduce(foldComplianceChangeSetToField, {})(changeSet));
```

Comment:
I kinda prefer the explicit because it's clear that it's an `array`, and easier to distinguish between that and a `tuple` @igbopie. but i have no beef with your linter